projects/canopy/src/lib/pipes/kebab-case/kebab-case.pipe.ts

```diff
@@ -7,6 +7,13 @@ export class LgKebabCasePipe implements PipeTransform {
   transform(str: string): any {
     return !str
       ? str
-      : str.toLowerCase().replace(/\s/g, '-');
+      : str
+        .replace(/[_\-|]{1,}/g, ' ')
+        .replace(/([a-z])([A-Z])/g, function(match, group1, group2) {
+          return `${group1} ${group2.toLowerCase()}`;
```

Comment:
Shouldn't the whole string be made lowercase?

Reply:
`group1` and `group2` here are just 2 characters(a lower case and an upper case, respectively), the whole thing is also lowercased at the end (`toLowerCase()`)